machinist/mnode/commands.go

```diff
@@ -0,0 +1,85 @@
+package mnode
+
+import (
+	"github.com/enfabrica/enkit/lib/client"
+	"github.com/enfabrica/enkit/machinist"
+	"github.com/spf13/cobra"
+)
+
+func NewRootCommand(bf *client.BaseFlags) *cobra.Command {
+
+	config := &Config{
+		bf:            bf,
+		ms:            &machinist.SharedFlags{},
+		enrollConfigs: &enrollConfigs{},
+	}
+
+	c := &cobra.Command{
+		Use: "node [OPTIONS] [SUBCOMMANDS]",
+	}
+	factory := func() (*Node, error) {
+		newN, err := New(config)
+		if err != nil {
+			return nil, err
+		}
+		return newN, err
+	}
+
+	// Global Relate Flags
+	c.PersistentFlags().StringArrayVar(&config.Tags, "tags", []string{}, "the list of tags you want this node to have. Setting this will unset the cache")
+	c.PersistentFlags().StringVar(&config.Name, "name", "no-name", "the name of this node. If a node already exists with this name, polling the machinist server will fail")
+	c.PersistentFlags().StringArrayVar(&config.DnsNames, "dns-names", []string{"localhost"}, "the list of dns names you want this node to have")
```

Comment:
uhm ... what happens when multiple hosts register as localhost? shouldn't this list be empty?

Also, I would call the flag something like `--aliases` or `--extra-names` ... dns is an implementation detail :) eg, allocation won't care about DNS, but will work by alias as well?

Reply:
So Dns Names is a misnomer, I was actually using these as ssh-names. Since this isn't in userland, I want it to be pretty specific for the operator. I renamed it to be more clear

We use localhost as a default because of the proxy, and because localhost is used for a lot of bastion type configs